annotate/module.ily

```diff
@@ -74,7 +74,8 @@
       (props (context-mod->props properties))
       ;; retrieve a pair with containing directory and input file
       (input-file (string-split (car (ly:input-file-line-char-column (*location*))) #\/ ))
-      (ctx (list-tail input-file (- (length input-file) 2)))
+      (ctx (list-tail input-file (- (length input-file)
+                                    (if (> (length input-file) 1) 2 1))))
```

Comment:
OK, I see the first effect: when your new switch applies `ctx` will be a list of *one* element (although without actually running it I can't tell what it will *be*). *Then* the following split will not work, `input-file-name` will contain an empty list (which is always the `cdr` of a one-element list).

OK, and the key to it is in line 96-98 (new state): `assq-set! props 'input-file-name` will assign this empty list to the prop, but - as stated in the comment - this is currently not used. Therefore no fatal side-effect is noticed.

---

I *assume* that in case of your fix `input-directory` will falsely be set to hold the input-filename, which will then be the default/fallback value for the "context". So - as said, without actually checking with running LilyPond - I would say what happens in the cause of failure is that "context" defaults to the file name instead of the directory, which will become visible when none of the other means of setting the context prop works. So this is the undetected potential side-effect.

But this idea of defaulting to the enclosing directory is somewhat weird anyway (or more concretely: it originated from the project I initially created this for where the directory name corresponded to the instrumental part), so I think I should thoroughly reconsider this construct anyway. Once more, could you please supply the value of `input-file` for the constellation where compilation aborted?